refactor(curriculum): functions quiz (#58381)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/25-front-end-development/quiz-javascript-functions/66edcc779993c0da6906dbb9.md

#### --distractors--

You should always omit the curly braces and `return` keyword.

---

You should never omit the curly braces and `return` keyword.

---

When the function body has multiple lines of code.

#### --answer--

When the function body consists of a single expression.

### --question--

#### --text--

What will be the result for the following code?

```js
function exampleFunction () {

}
exampleFunction();
```

#### --distractors--

The program will crash.

---

JavaScript will skip over that piece of code.

---

An error message will display.

#### --answer--

Nothing will happen.

### --question--

#### --text--

Which of the following is the correct way to use default parameters?

#### --distractors--

```js
function sum (num1, num2 <<>> 12) {
  return num1 + num2
} 
```

---

```js
function sum (num1, num2 === 12) {
  return num1 + num2
} 
```

---

```js
function sum (num1, num2 >> 12) {
  return num1 + num2
} 
```

#### --answer--

```js
function sum (num1, num2 = 12) {
  return num1 + num2
} 
```

### --question--

#### --text--

What will be the result for the following code?

```js
const developer = "Jessica";

> function greet(){

Reviewer: ```suggestion
function greet() {
```

  console.log("Hello, " + developer)
}

greet();
```

#### --distractors--

> The string `"Hello, developer"` is logged to the console.

Reviewer: ```suggestion
The string `"Hello, developer"` will be logged to the console.
```

---

An error message will display in the console.

---

Nothing will display in the console.

#### --answer--

> The string `"Hello, Jessica"` is logged to the console.

Reviewer: ```suggestion
The string `"Hello, Jessica"` will be logged to the console.
```

### --question--

#### --text--

What happens if you try to remove the curly braces for a regular function?

#### --distractors--

An alert box will display on the page with an error.

---

The program will crash and not start again.

---

Nothing will happen.

#### --answer--

You will get a syntax error.

### --question--

#### --text--

What will be the result for the following code?

```js
const sum = (num1, num2) => num1 + num2

> console.log(sum(0,0) + 10);

Reviewer: ```suggestion
console.log(sum(0, 0) + 10);
```

```

#### --distractors--

An error message will display in the console.

---

> The number 0 will be logged to the console.

Reviewer: ```suggestion
The number `0` will be logged to the console.
```

---

Nothing will be logged to the console.

#### --answer--

> The number 10 will be logged to the console.

Reviewer: ```suggestion
The number `10` will be logged to the console.
```

#### --text--

What will be the result for the following code?

```js
const sum = (num1, num2) => num1 + num2

> console.log(sum(0,0) + num2);

Reviewer: ```suggestion
console.log(sum(0, 0) + num2);
```

```

#### --distractors--

An error message saying `num1` is not defined.

---

> The number 0 will be displayed in the console.

Reviewer: ```suggestion
The number `0` will be displayed in the console.
```
